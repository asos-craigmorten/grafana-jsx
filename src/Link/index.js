import PropTypes from "prop-types";

const Link = ({
  icon = "external link",
  tags = [],
  type = "dashboards",
  ...props
}) => <link icon={icon} tags={tags} type={type} {...props} />;

Link.PropTypes = {
  icon: PropTypes.string,
  tags: PropTypes.arrayOf(PropTypes.string),
  type: PropTypes.string,
};

export default Link;
