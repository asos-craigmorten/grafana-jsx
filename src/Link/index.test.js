import Link from "./";

const withAllProps = {
  icon: "test-icon",
  tags: ["test-tag-1", "test-tag-2"],
  type: "test-type",
};

describe("Link", () => {
  describe("when passed no props", () => {
    it("should render a link with defaults", () => {
      expect(<Link />).toMatchSnapshot();
    });
  });

  describe("when passed all props", () => {
    it("should render a custom link", () => {
      expect(<Link {...withAllProps} />).toMatchSnapshot();
    });
  });

  describe("when passed custom props", () => {
    it("should render a custom link with the custom props", () => {
      expect(
        <Link {...withAllProps} test-link-prop={Symbol("test-link-value")} />
      ).toMatchSnapshot();
    });
  });
});
