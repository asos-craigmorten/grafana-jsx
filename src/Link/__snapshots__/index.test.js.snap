// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link when passed all props should render a custom link 1`] = `
Object {
  "icon": "test-icon",
  "tags": Array [
    "test-tag-1",
    "test-tag-2",
  ],
  "type": "test-type",
}
`;

exports[`Link when passed custom props should render a custom link with the custom props 1`] = `
Object {
  "icon": "test-icon",
  "tags": Array [
    "test-tag-1",
    "test-tag-2",
  ],
  "test-link-prop": Symbol(test-link-value),
  "type": "test-type",
}
`;

exports[`Link when passed no props should render a link with defaults 1`] = `
Object {
  "icon": "external link",
  "tags": Array [],
  "type": "dashboards",
}
`;
